File: Homegrown Problems/Trond trenger hjelp/brute_force.hs
import           Control.Arrow ((>>>))
import           Data.Char     (isSpace)
import           Data.Functor  ((<&>))

input :: String -> IO [Int]
input file = readFile file <&> (dropWhile (/='[') >>> filter (isSpace >>> not) >>> read)

solve :: [Int] -> Int
solve []     = 0
solve [x]    = x
solve (x:xs) = max (x + solve (drop 1 xs)) (solve xs)

File: Homegrown Problems/Trond trenger hjelp/brute_force.py
from data import data


def brute_force(i):
    if i < 0:
        return 0
    if i == 0:
        return data[i]
    else:
        return max(brute_force(i - 1), brute_force(i - 2) + data[i])


print(brute_force(len(data) - 1))
